app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,20 @@
+<h1><%= @recipe.name %></h1>
+<h2>Complexity: <%= @recipe.complexity %></h2>
+<h2>Genre: <%= @recipe.genre %></h2>
+<% @recipe_ingredients.each do |recipe_ingredient| %>
+  <ul>
+    <%= recipe_ingredient.ingredient.name %>
+  </ul>
+<% end %>
+<h3>Total Cost: $<%= @recipe.total_cost %></h3>
+<br>
+<h3>Add Ingredient to <%= @recipe.name %></h3>
+<%= form_with url: "/recipes/#{@recipe.id}", method: :patch, local: true do |f| %>
```

Comment:
This is close! The 2nd extension asks us to create a new relationship between an existing recipe (from this show page) and an ingredient by its id. As such, the path would not be to a recipe, but rather to either
```
/recipe/:id/ingredients
```
or
```
/recipe_ingredients
```
since those are the relationships or models we'll be affecting the most. 
Also, since we're *creating* a new relationship, we'll use the method `:post`.